app/src/main/java/study_dev/testbed/studyhopper/CreateGroup.java

```diff
@@ -0,0 +1,283 @@
+import android.widget.ImageView;
+import android.widget.NumberPicker;
+import android.widget.Spinner;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+public class CreateGroup extends AppCompatActivity implements View.OnClickListener, AdapterView.OnItemSelectedListener {
+    private EditText editTextGroupName;
+    private EditText editTextCourseCode;
+    private String groupColor, groupPreference;
+    private Spinner groupPreferencesSpinner;
+    private NumberPicker groupMaxSizePicker;
+    private EditText editTextMaxSize;
+    private TextView groupPreferencesPrompt;
+    private int colorSelected = 0;
+    private ImageView blue, green, yellow, red, purple, orange, brown, gray;
+
+    private FirebaseAuth mAuth = FirebaseAuth.getInstance();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_create_group);
+
+        editTextGroupName = findViewById(R.id.edit_text_group_name);
+        editTextCourseCode = findViewById(R.id.edit_text_group_course_code);
+
+        editTextMaxSize = findViewById(R.id.edit_text_group_max);
+        blue = findViewById(R.id.image_view_blue);
+        green = findViewById(R.id.image_view_green);
+        yellow = findViewById(R.id.image_view_yellow);
+        red = findViewById(R.id.image_view_red);
+        purple = findViewById(R.id.image_view_purple);
+        orange = findViewById(R.id.image_view_orange);
+        brown = findViewById(R.id.image_view_brown);
+        gray = findViewById(R.id.image_view_gray);
+
+        blue.setOnClickListener(this);
+        green.setOnClickListener(this);
+        yellow.setOnClickListener(this);
+        red.setOnClickListener(this);
+        purple.setOnClickListener(this);
+        orange.setOnClickListener(this);
+        brown.setOnClickListener(this);
+        gray.setOnClickListener(this);
+
+        getSupportActionBar().setHomeAsUpIndicator(R.drawable.ic_close);
+        setTitle("Create Group");
+
+        // Enable back button
+        ActionBar supportActionBar = getSupportActionBar();
+        if (supportActionBar != null) {
+            supportActionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        groupPreferencesSpinner = findViewById(R.id.spinner_group_preferences);
+        fillGroupPreferencesSpinner();
+    }
+
+    public void createStudyGroup(View v) {
+        String groupName = editTextGroupName.getText().toString();
+        String courseCode = editTextCourseCode.getText().toString();
+        String groupPreference = groupPreferencesSpinner.getSelectedItem().toString();
+        int groupMaxSize = Integer.parseInt(editTextMaxSize.getText().toString());
+        boolean coedGroup = false, femalesOnlyGroup = false, malesOnlyGroup = false;
```

Comment:
this should probably be a single field, like an int. saving all 3 values when only 1 is valid is redundant

Reply:
Changed the groupPreferences to a single string field. Modified the Groups.java class accordingly.